modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,480 @@
+
+			RenTokenRegistry::<T>::remove(&_ren_token_id);
+
+			Self::deposit_event(RawEvent::RenTokenDeleted(_ren_token_id));
+			Ok(())
+		}
+
+
+		/// Transfers tokens from the pallet account to a specified account
+		#[weight = T::WeightInfo::spend_tokens()]
+		pub fn spend_tokens(
+			origin,
+			#[compact] _ren_token_id: TokenIdOf<T>,
+			who: T::AccountId,
+			amount: BalanceOf<T>,
+		) -> DispatchResult
+		{
+			T::ControllerOrigin::ensure_origin(origin)?;
+
+			ensure!(<RenTokenRegistry<T>>::contains_key(&_ren_token_id), Error::<T>::RenTokenNotFound);
+
+			T::Assets::transfer(_ren_token_id, Self::account_id().into(), who.clone(), amount)?;
+
+			Self::deposit_event(RawEvent::RenTokenSpent(_ren_token_id, who, amount));
+			Ok(())
+		}
+
+		/// Mints the token, after fee deduction, if the signature provided is valid over the provided arguments
+		#[weight = T::WeightInfo::validate_and_mint()]
+		pub fn mint(
+			origin,
+			#[compact] _ren_token_id: TokenIdOf<T>,
+			who: T::AccountId,
+			p_hash: [u8; 32],
+			amount: BalanceOf<T>,
+			n_hash: [u8; 32],
+			sig: EcdsaSignature,
+		) -> DispatchResult
+		{
+			ensure_none(origin)?;
+
+			let ren_token = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?;
+
+			ensure!(ren_token.ren_token_mint_enabled, Error::<T>::RenTokenMintDisabled);
+
+			let mint_fee_value = Permill::from_parts(ren_token.ren_token_mint_fee).mul_floor(amount);
+
+			// Skipped checking if the mint call for fee might overflow or cause min_balance error
+			// as that should not prevent a user from his operation
+
+			// MINT CALL for user
+			T::Assets::mint(_ren_token_id, who.clone(), (amount - mint_fee_value).into())?;
+			// Attempt mint call for fees
+			let _ = T::Assets::mint(_ren_token_id, Self::account_id().into(), mint_fee_value.into());
+
+			Signatures::insert(&sig, ());
+			Self::deposit_event(RawEvent::RenTokenMinted(_ren_token_id, who, amount));
+			Ok(())
+		}
+
+
+		/// Burns the token so that the corresponding pegged currency can be released to the "to" address by RenVM
+		/// Checks if the token is bitcoin by name, "renBTC" (or "renTestBTC" for testnet) and validates the "to" address accordingly
+		/// The actual amount burnt is calculated based on the burn fee, and is the value of the corresponding pegged currency that the user will be granted from RenVM
+		/// The burn fee is transfered from the users account to the pallets account
+		#[weight = T::WeightInfo::burn()]
+		pub fn burn(
+			origin,
+			#[compact] _ren_token_id: TokenIdOf<T>,
+			to: DestAddress,
+			amount: BalanceOf<T>,
+		) -> DispatchResult
+		{
+			let sender = ensure_signed(origin)?;
+
+			let ren_token = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?;
+
+			ensure!(ren_token.ren_token_burn_enabled, Error::<T>::RenTokenBurnDisabled);
+
+			// match sp_std::str::from_utf8(ren_token.ren_token_name.as_slice()).map_err(|_| Error::<T>::UnexpectedError) {
```

Comment:
The library you mentioned is blocking integrating this right?

Reply:
Yes.
Coinaddress seems to be quite popular. I have not been able to find an alternative nor have I been able to resolve the issue.